runners/samza/src/main/java/org/apache/beam/runners/samza/translation/ConfigContext.java

```diff
@@ -35,12 +34,13 @@ public class ConfigContext {
   private final Map<PValue, String> idMap;
   private AppliedPTransform<?, ?, ?> currentTransform;
   private final SamzaPipelineOptions options;
-  private final Set<String> stateIds;
+  private final Set<String> nonUniqueStateIds;
```

Comment:
Please take a look at the comments below to wrap this in a StoreIdGenerator class.

Reply:
Done.